velox/expression/ExprToSubfieldFilter.cpp

```diff
@@ -477,47 +478,47 @@ PrestoExprToSubfieldFilterParser::leafCallToSubfieldFilter(
 
   const auto* leftSide = call.inputs()[0].get();
 
-  if (call.name() == "eq") {
+  if (call.name() == expression::bug::kEq) {
```

Comment:
I think we can add a functionality to evaluator to resolve function names and return some canonical names, so we can do these expression transformation for well known functions.

Reply:
Maybe such approach can be used?

```
bool isNotExpr(
    const core::TypedExprPtr& expr,
    const core::CallTypedExpr* call,
    core::ExpressionEvaluator* evaluator) {
  if (!endWith(call->name(), "not")) {
    return false;
  }
  auto exprs = evaluator->compile(expr);
  VELOX_CHECK_EQ(exprs->size(), 1);
  auto& compiled = exprs->expr(0);
  return compiled->vectorFunction() &&
      compiled->vectorFunction()->getCanonicalName() ==
      exec::FunctionCanonicalName::kNot;
}

```